contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -43,11 +57,133 @@ contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
+
+        if (rebalanceCheck()) {
+            // 0. Get tickspacing from vamm
+            // int24 _tickSpacing = _vamm.tickSpacing(_vamm.address());
+
+            // 1. Get the new tick lower
+            // uint256 _newFixedLowerWad = Math.min(Math.max(0, uint256(currentFixedRateWad) - _sigmaWad), _max_possible_lower_bound);
+            uint256 deltaWad = uint256(currentFixedRateWad) - _sigmaWad;
+            console.log(deltaWad);
+            uint256 _newFixedLowerWad =  0;
+            if (deltaWad > 0) {
+                // delta is greater than 0 => choose delta
+                if (deltaWad < _max_possible_lower_bound) {
+                    _newFixedLowerWad = deltaWad;
+                } else {
+                    _newFixedLowerWad = _max_possible_lower_bound;
+                }
+            } else {
+                // delta is less than 0 => choose 0
+                if (_max_possible_lower_bound > 0) {
+                    _newFixedLowerWad = 0;
+                } else {
+                    _newFixedLowerWad = _max_possible_lower_bound;
+                }
+            }
+            // 2. Get the new tick upper
+            console.log(_newFixedLowerWad);
+            uint256 _newFixedUpperWad = _newFixedLowerWad + 2 * _sigmaWad;
+            console.log(_newFixedUpperWad);
+            // 3. Convert new fixed lower rate back to tick
+            int256 _newTickLowerWad = -PRBMathSD59x18.div(PRBMathSD59x18.log2(int256(_newFixedUpperWad)), 
+                                                        PRBMathSD59x18.log2(1000100000000000000)
+                                                        );
+            console.logInt(_newTickLowerWad);
+            // 4. Convert new fixed upper rate back to tick
+            int256 _newTickUpperWad = -PRBMathSD59x18.div(PRBMathSD59x18.log2(int256(_newFixedLowerWad)),
+                                                        PRBMathSD59x18.log2(1000100000000000000)
+                                                        );
+            console.logInt(_newTickUpperWad);
+
+            int256 _newTickLower = _newTickLowerWad/1e18;
+            int256 _newTickUpper = _newTickUpperWad/1e18;
+
+            console.logInt(_newTickLower);
+            console.logInt(_newTickUpper);
+
+            int24 _newTickLowerMul = nearestTickMultiple(int24(_newTickLower), _tickSpacing);
+            int24 _newTickUpperMul = nearestTickMultiple(int24(_newTickUpper), _tickSpacing);
+
+            console.logInt(_newTickLowerMul);
+            console.logInt(_newTickUpperMul);
+
+            emit rebalanced(_newTickLowerMul, _newTickUpperMul);
+            return (_newTickLowerMul, _newTickUpperMul);
+        } else {
+            revert(ExceptionsLibrary.REBALANCE_NOT_NEEDED);
+          }
+        }
+
+
+//--------------------// Don't need the below functions // ---------------------//
+
     /// @notice Get new signal and act according to it
```

Comment:
need to remove all functions downwards. they are trading vault specific

Reply:
also remove ILPCallback inheritance after you do this.

Reply:
I have removed these